inference-engine/src/mkldnn_plugin/nodes/mkldnn_gemm_node.cpp

```diff
@@ -122,8 +122,13 @@ void MKLDNNGemmNode::initSupportedPrimitiveDescriptors() {
     auto inPrec0 = getCnnLayer()->insData[0].lock()->getPrecision();
     auto inPrec1 = getCnnLayer()->insData[1].lock()->getPrecision();
     if ((inPrec0 != Precision::U8 && inPrec0 != Precision::I8) || inPrec1 != Precision::I8 || isThreeInputs) {
-        inPrec0 = Precision::FP32;
-        inPrec1 = Precision::FP32;
+        if ((inPrec0 == Precision::BF16 || inPrec1 == Precision::BF16) && !isThreeInputs) {
```

Comment:
From my understanding we should allow BF16 inference only in case both inputs are BF16. The node shouldn't decrease precision itself.

Reply:
Not all of layers work on BF16. Here we are ordering a 'Reorder' for non BF16 input branch